### DISCUSSION TITLE
why milvus uses its own etcd image instead of the one provided by bitnami?

### DISCUSSION THREAD

**Discussion by caozhuozi** [milvus-io/milvus#27707] [Q&A and General discussion]
I observed that there is no prestop hook in  the milvus-based etcd image:

 **prestop.sh is empty!**

```shell
I have no name!@milvus230-etcd-0:/opt/bitnami/scripts/etcd$ cat /opt/bitnami/scripts/etcd/prestop.sh
#!/bin/bash

# do nothing
```
Whats the purpose of this?

Could someone help answer this question?

**Comment by LoveEachDay** [comment]
We have customized the startup procedures for etcd. If you do care, you can firstly set up a etcd cluster, then setup a milvus cluster with external etcd cluster.
